src/main.js

```diff
@@ -44,6 +47,27 @@ if (!customElements.get('ha-icon')) {
   );
 }
 
+function luminance(r, g, b) {
```

Comment:
Move all color related util functions into its own file under `/utils/` and import them in main.